src/main/kotlin/com/codingchallenge/transactionanalyser/TransactionAnalyserApp.kt

```diff
@@ -0,0 +1,29 @@
+package com.codingchallenge.transactionanalyser
+
+import java.lang.IllegalArgumentException
+import java.time.format.DateTimeFormatter
+
+class TransactionAnalyserApp {
+    fun runApp(){
+        val transactionLoader = TransactionLoader(
+            this::class.java.classLoader.getResourceAsStream("transactions.csv") ?: throw IllegalArgumentException("File not found")
```

Comment:
Just an FYI.
Ensure you are familiar with the Exceptions best practices in kotlin. They are different to in Java. Obviously Java has checked exceptions and unchecked. Kotlin does not have checked exceptions.

https://elizarov.medium.com/kotlin-and-exceptions-8062f589d07 Roman Elizarov is on the Kotlin team and used to be the tech lead of the language team itself. His post here is very insightful on where we should be considering exception handling in our Kotlin code.
Think about code you have seen where there are `try`/`catch` blocks polluting an entire chain of functions. And consider what he is saying here to keep that in one place, and let the exception bubble up gracefully.

Reply:
It will be wise to follow Roman Elizarov on Twitter as well

Reply:
> Just an FYI.
> Ensure you are familiar with the Exceptions best practices in kotlin. They are different to in Java. Obviously Java has checked exceptions and unchecked. Kotlin does not have checked exceptions.
> 
> https://elizarov.medium.com/kotlin-and-exceptions-8062f589d07 Roman Elizarov is on the Kotlin team and used to be the tech lead of the language team itself. His post here is very insightful on where we should be considering exception handling in our Kotlin code.
> Think about code you have seen where there are `try`/`catch` blocks polluting an entire chain of functions. And consider what he is saying here to keep that in one place, and let the exception bubble up gracefully.

Will have a closer look at how Kotlin handles exceptions.